framework/src/problems/FEProblemBase.C

```diff
@@ -4179,6 +4184,348 @@ FEProblemBase::solve()
     _displaced_problem->syncSolutions();
 }
 
+bool
+FEProblemBase::newSolve()
+{
+  TIME_SECTION(_solve_timer);
+
+  const auto & solver_params = solverParams();
+
+  Real current_dt = dt();
+
+  // these variables can just be local, but because we need to keep them for reporting
+  // at timestep end, we make them member variables so that postprocessors can see them.
+  // They really are modified locally!
+  unsigned int & picard_it = _picard_it;
+  Real & picard_initial_norm = _picard_initial_norm;
+  std::vector<Real> & picard_timestep_begin_norm = _picard_timestep_begin_norm;
+  std::vector<Real> & picard_timestep_end_norm = _picard_timestep_end_norm;
+
+  picard_timestep_begin_norm.resize(solver_params._picard_max_its);
+  picard_timestep_end_norm.resize(solver_params._picard_max_its);
+
+  bool converged = true;
+  _picard_status = MOOSE_PICARD_ITERATING;
+
+  // need to back up multi-apps even when not doing Picard iteration for recovering from failed
+  // multiapp solve
+  backupMultiApps(EXEC_TIMESTEP_BEGIN);
+  backupMultiApps(EXEC_TIMESTEP_END);
+
+  // Prepare to relax variables as a master
+  std::set<dof_id_type> relaxed_dofs;
+  if (solver_params._picard_relaxation_factor != 1.0)
+  {
+    // Snag all of the local dof indices for all of these variables
+    System & libmesh_nl_system = _nl->system();
+    AllLocalDofIndicesThread aldit(libmesh_nl_system, solver_params._picard_relaxed_variables);
+    ConstElemRange & elem_range = *mesh().getActiveLocalElementRange();
+    Threads::parallel_reduce(elem_range, aldit);
+
+    relaxed_dofs = aldit._all_dof_indices;
+  }
+
+  // Prepare to relax variables as a subapp
+  std::set<dof_id_type> self_relaxed_dofs;
+  if (solverParams()._picard_self_relaxation_factor != 1.0)
+  {
+    // Snag all of the local dof indices for all of these variables
+    System & libmesh_nl_system = _nl->system();
+    AllLocalDofIndicesThread aldit(libmesh_nl_system,
+                                   solverParams()._picard_self_relaxed_variables);
+    ConstElemRange & elem_range = *mesh().getActiveLocalElementRange();
+    Threads::parallel_reduce(elem_range, aldit);
+
+    self_relaxed_dofs = aldit._all_dof_indices;
+
+    if (previousTimeAsSubapp() == _time)
+    {
+      NumericVector<Number> & solution = _nl->solution();
+      NumericVector<Number> & relax_previous = _nl->getVector("self_relax_previous");
+      relax_previous = solution;
+    }
+  }
+
+  picard_it = 0;
+  while (picard_it < solver_params._picard_max_its)
+  {
+    if (solver_params._has_picard_its)
+    {
+      _console << COLOR_MAGENTA << "Beginning Picard Iteration " << picard_it << COLOR_DEFAULT
+               << '\n';
+
+      if (picard_it ==
+          0) // First Picard iteration - need to save off the initial nonlinear residual
```

Comment:
This wrapping is odd, perhaps move the comment up above the if statement?

Reply:
wow, this is done by the formatter. I will move the inline comments down inside the if branch.